aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -180,12 +201,57 @@ impl ProviderConfig {
         self.region.clone()
     }
 
+    pub(crate) async fn try_profile(&self) -> Result<&ProfileSet, &ProfileFileLoadError> {
+        let parsed_profile = self
+            .parsed_profile
+            .get_or_init(|| async {
+                let profile = profile::load(
+                    &self.fs,
+                    &self.env,
+                    &self.profile_files,
+                    self.profile_name_override.clone(),
+                )
+                .await;
+                if let Err(err) = profile.as_ref() {
+                    tracing::warn!(err = %DisplayErrorContext(&err), "failed to parse profile")
+                }
+                profile
+            })
+            .await;
+        parsed_profile.as_ref()
+    }
+
+    pub(crate) async fn profile(&self) -> Option<&ProfileSet> {
+        self.try_profile().await.ok()
+    }
+
     /// Override the region for the configuration
     pub fn with_region(mut self, region: Option<Region>) -> Self {
         self.region = region;
         self
     }
 
+    /// Override profile-file (`~/.aws/config` et al) configuration
```

Comment:
I think we should avoid conflating the path, the files, and the config info extracted from one or more files. 

```suggestion
    /// Override the profile file paths (`~/.aws/config` by default) and name (`default` by default)
```